### PR TITLE
dwarf-fortress: 0.47.04 -> 0.47.05

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Can't get the game to launch without seg faulting

maybe @shazow @bbigras  you could help?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114710` at b936d598 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/d1a2dbce7b96e2d3e8c4715c163ab512">dwarf-fortress-packages.dwarf-therapist-original</a></li></li>
  </ul>
</details>

**Comment by tomberek** [comment]
I have to go into PRINT_MODE:TEXT to avoid an SDL initialization failure. I'm running on wayland/sway if that matters.

**Comment by plabadens** [comment]
> I have to go into PRINT_MODE:TEXT to avoid an SDL initialization failure. I'm running on wayland/sway if that matters.

That's probably because you have `SDL_VIDEODRIVER=wayland` in your environment. The SDL version used by dwarf-fortress is too old to have heard of Wayland. You should be able to use `SDL_VIDEODRIVER=xcb` when launching dwarf-fortress.

**Comment by plabadens** [comment]
Would it make sense to update dwarf-therapist separately (to at least `v41.2.0`)? The current version is incompatible with the latest version of dwarf-fortress (`v47.0.4`) in nixpkgs.

**Comment by jonringer** [comment]
> Would it make sense to update dwarf-therapist separately (to at least `v41.2.0`)? The current version is incompatible with the latest version of dwarf-fortress (`v47.0.4`) in nixpkgs.

I can do that

**Comment by jonringer** [comment]
dwarf-therapist was updated: https://github.com/NixOS/nixpkgs/pull/127136

**Comment by jonringer** [comment]
I'm not sure why, but launching the game with gdb, I was able to determine that the crash was coming from unfuck. For now, I just chose to alias the 0.47.05 to 0.47.04 release, and it seems to have fixed everything... (probably gtk3 related failures, not sure)

**Comment by jonringer** [comment]
cc @Artturin

**Review by Artturin — APPROVED** [review]
i dont actually play dwarf-fortress but the pr builds and the game launches

**Comment by jonringer** [comment]
I've been playing it for a while.

File: pkgs/games/dwarf-fortress/twbt/default.nix [hunk 1]
```diff
@@ -50,6 +50,12 @@ let
       sha256 = "092dgp8fh1j4nqr9wbzn89ib1nhscclr8m91lfxsvg0mgn7j8xlv";
       prerelease = true;
     };
+    "0.47.05" = {
+      twbtRelease = "6.xx";
+      dfhackRelease = "0.47.05-beta1";
```
Comment (kevincox) [inline pkgs/games/dwarf-fortress/twbt/default.nix:55]: This seems to cause problems because it doesn't match "0.47.05-r1" above.

Comment (kevincox) [reply pkgs/games/dwarf-fortress/twbt/default.nix:55]: The fix seems easy: https://github.com/NixOS/nixpkgs/pull/132263

**Pull request merged**